app/test/integration_test.go

```diff
@@ -349,11 +352,16 @@ func (s *IntegrationTestSuite) TestShareInclusionProof() {
 
 // ExtendBlobTest re-extends the block and compares the data roots to ensure
 // that the public functions for extending the block are working correctly.
-func (s *IntegrationTestSuite) ExtendBlobTest(t *testing.T, block *coretypes.Block) {
+func ExtendBlobTest(t *testing.T, block *coretypes.Block) {
 	eds, err := app.ExtendBlock(block.Data, block.Header.Version.App)
 	require.NoError(t, err)
 	dah := da.NewDataAvailabilityHeader(eds)
-	require.Equal(t, dah.Hash(), block.DataHash.Bytes())
+	if !assert.Equal(t, dah.Hash(), block.DataHash.Bytes()) {
+		// save block to json file for further debugging if this occurs
+		b, err := json.MarshalIndent(block, "", "  ")
+		require.NoError(t, err)
+		require.NoError(t, ioutil.WriteFile(fmt.Sprintf("bad_block_%s.json", tmrand.Str(6)), b, 0644))
```

Comment:
[question] is this strictly for local debugging? Do we have access to this artifact if it occurs on CI?

Reply:
yeah, this is only for local cases, but if it ever occurs then we automatically have a saved instance of the block so decided to keep this

its totally optional tho, so we can remove if we don't find it valuable